compiler/vm/vm.nim

```diff
@@ -73,26 +73,44 @@ const
 
 proc stackTraceImpl(
     c:          TCtx,
-    sframe:     PStackFrame,
+    sframe:     StackFrameIndex,
     pc:         int,
     lineInfo:   TLineInfo,
     infoOrigin: InstantiationInfo,
     recursionLimit: int = 100
   ) =
-
-  proc aux(c: TCtx, sframe: PStackFrame, pc, depth: int, res: var SemReport) =
-    if sframe != nil:
+  ## Generate and report the stack trace starting at frame `sframe` (inclusive)
+  assert c.sframes.len > 0
+
+  # TODO: use the below once exception handling is reworked, and stack frames 
+  # don't form a tree any longer
+  #[
+  let bottom = high(c.sframes)
+  let top = max(0, bottom - recursionLimit)
+  
+  # TODO: Skipped number of entries don't get reported? How is/was this intended to be handled?
+  # let numSkipped = top
+  for i in top..bottom:
+    let f = c.sframes[i]
+    let fpc = if i < bottom: c.sframes[i+1].comesFrom else: pc
+    res.stacktrace.add((sym: f.prc, location: c.debug[fpc]))
+  ]#
+
+  proc aux(c: TCtx, sframe: StackFrameIndex, pc, depth: int, res: var SemReport) =
+    if sframe != -1:
       if recursionLimit < depth:
+        # TODO: Skipped number of entries don't get reported? How is/was this intended to be handled?
```

Comment:
As far as I can tell, stack frames with a depth below the the recursion limit get skipped here, but aren't getting reported

Reply:
Yes, it's for pruning